commands.js

```diff
@@ -68,7 +69,96 @@ const create_withdraw_request = async (name, amount) => {
     }
 }
 
+async function loop_check_bulk_transfer(scan_request_id) {
+    return new Promise((resolve, reject) => {
+        const intervalId = setInterval(async () => {
+            try {
+                const info = await get_scan_request({ scan_request_id });
+                console.log(`Scan request with id: ${scan_request_id} has status: ${info.status}`);
+                
+                if (info.status === 'FAILED_LIMITS_EXCEEDED') {
+                    const user_limits_message = await validate_user_limits();
+                    console.log(user_limits_message);
+                    clearInterval(intervalId); // Stop the interval before throwing an error
+                    throw new Error(user_limits_message);
+                }
+
+                if (info.status === 'FAILED') {
+                    const message = `Scan request with id: ${scan_request_id} failed`;
+                    console.log(message);
+                    clearInterval(intervalId); // Stop the interval before throwing an error
+                    throw new Error(message);
+                }
+
+                if (info.status === 'COMPLETED') {
+                    clearInterval(intervalId); // Stop the interval
+                    save_bulk_transfer(scan_request_id, info);
+                    resolve(); // Resolve the promise
+                }
+            } catch (err) {
+                console.log(`Error checking bulk transfer with id: ${scan_request_id}: ${err.message}`);
+                clearInterval(intervalId); // Ensure the interval is cleared on error
+                reject(err); // Reject the promise
+            }
+        }, 1000); // Check every second
+    });
+}
+
+const bulk_transfer = async (p_from_address, p_to_address, p_tag_to_extract, p_num_of_tag_to_send, p_fee_rate) => {
+    try {
```

Comment:
@habibitcoin @topether21 
Can we send a telegram notification here that just says: "Scan has been initiated, see terminal logs to track progress..." or something

(I'm running a scan on our many thousand utxos and its taking a while)

Reply:
Yup, we'll add as a follow up